### PR TITLE
Add ability to set gomodifytag options in config

### PR DESCRIPTION
**Pull request by svrana**

### CODE REVIEW

File: lua/go/tags.lua [hunk 1]
```diff
@@ -34,17 +36,27 @@ tags.modify = function(...)
...
   local arg = { ... }
   local transflg = false
+  local optsflg = false
   for _, v in ipairs(arg) do
     table.insert(setup, v)
     if v == "-transform" or v == "-t" then
       transflg = true
     end
+    if v == "-add-options" then
+      optsflg = true
```
Comment (ray-x) [inline lua/go/tags.lua:46]: How about allowing the user to pass in the options here 
e.g. 
`GoModifyTags -add-option json=omitempty`

**Review by ray-x — COMMENTED** [review]
Thanks for the PR.
json=omitempty can be a default
But we should allow users to specify parameters to override default settings.

**Comment by svrana** [comment]
Forced pushed some changes:
   default tag_options to json=omitempty
   fixed indent level allowing overwriting of default tag options

Tested the default config and omitempty was added. Tested with 'GoAddTag json -add-options json=' and json tag was added without the omitempty. Tested by modifying tag-options to 'json='  in my nvim config and confirmed omitempty was not added in the tag.

**Review by ray-x — APPROVED** [review]
thanks!

**Pull request merged**